### PR TITLE
Annotations file arrangement

### PR DESCRIPTION
**Pull request by wild-endeavor**
* Added a new `create_and_link_node` function in node.py to replace the compile method in task, launch plan, and workflow.
* Fix the file handling behavior
* Removed the `native_kwargs_to_literal_map` in launch plan cuz I realized we can use the `translate_inputs_to_literals` function instead.
* Added a ReferenceTask, no workflows or launch plans yet.
* Started moving things around.  Removed all the flytekit imports from setup.py and replaced with literals - we can work out a better solution in the future, maybe ask around for suggestions.
* Moved logger out of flytekit init into a separate file, but imported it back into init, along with most things users will need.
* Made typing a folder and moved flytefile into a file.  Renamed `FlyteFilePath` to `FlyteFile`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,7 +32,9 @@ lint: ## Run linters
 
 .PHONY: test
 test: lint ## Run tests
-	pytest tests/flytekit/unit
+	pytest tests/flytekit/unit -k "not test_task"
```
Comment (wild-endeavor) [inline Makefile:35]: will remove before merging.

File: tests/flytekit/unit/models/test_dynamic_job.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 ]
 
 
-@pytest.mark.parametrize("task", LIST_OF_DYNAMIC_TASKS)
+@pytest.mark.parametrize("task", LIST_OF_DYNAMIC_TASKS[0:100])
```
Comment (wild-endeavor) [inline tests/flytekit/unit/models/test_dynamic_job.py:32]: will remove before merging

File: flytekit/annotated/type_engine.py [hunk 3]
```diff
@@ -368,64 +368,103 @@ def to_literal(
...
+        if not ctx.file_access.is_remote(uri):
+            return uri
 
+        # Since no delayed downloading is possible with strings, always download immediately.
+        ctx.file_access.get_data(lv.scalar.blob.uri, local_destination_path, is_multipart=False)
+        return local_destination_path
 
-class FlyteFilePathTransformer(TypeTransformer[flyte_typing.FlyteFilePath]):
+
+class FlyteFilePathTransformer(TypeTransformer[flytekit.typing.flyte_file.FlyteFile]):
```
Comment (kumare3) [inline flytekit/annotated/type_engine.py:396]: should this be kept next to the `typing` code?

**Comment by kumare3** [comment]
lgtm

**Pull request merged**